openmc/deplete/microxs.py

```diff
@@ -192,8 +210,118 @@ def __init__(self, data: np.ndarray, nuclides: List[str], reactions: List[str]):
         self._index_nuc = {nuc: i for i, nuc in enumerate(nuclides)}
         self._index_rx = {rx: i for i, rx in enumerate(reactions)}
 
-    # TODO: Add a classmethod for generating MicroXS directly from cross section
-    # data using a known flux spectrum
+    @classmethod
+    def from_multi_group_flux(
+        cls,
+        energies: Union[Iterable[float], str],
+        multi_group_flux: Sequence[float],
+        chain_file: Optional[PathLike] = None,
+        openmc_data_path: Optional[PathLike] = None,
+        temperature: int=294
+    ):
+        """Generated MicroXS object from a known flux and a chain file.
+
+        The size of the MicroXS matrix depends on the chain file and cross
+        sections available. MicroXS entry will be 0 if the nuclide cross section
+        is not found.
+
+        Parameters
+        ----------
+        energies : iterable of float or str
+            Energy group boundaries in [eV] or the name of the group structure
+        multi_group_flux : iterable of float
+            Energy-dependent multigroup flux values
+        chain_file : str, optional
+            Path to the depletion chain XML file that will be used in
+            depletion simulation.
+        openmc_data_path : str, optional
+            Path to the cross section XML file that contains data library paths.
+        temperature : int, optional
+            Temperature for cross section evaluation in [K].
+
+        Returns
+        -------
+        MicroXS
+        """
+
+        check_type("temperature", temperature, int)
+        # if energy is string then use group structure of that name
+        if isinstance(energies, str):
+            energies = openmc.EnergyFilter.from_group_structure(energies).values
+        else:
+            # if user inputs energies check they are ascending (low to high) as some
+            # depletion codes use high energy to low energy.
+            if not all(energies[i] <= energies[i+1] for i in range(len(energies) - 1)):
+                raise ValueError('Energy bin must be in ascending order')
+
+        # check dimension consistency
+        if not len(multi_group_flux) == len(energies)-1:
+            raise ValueError('Length of flux array should be len(energies)-1')
+
+        chain_file_path = _resolve_chain_file_path(chain_file)
+        chain = openmc.deplete.Chain.from_xml(chain_file_path)
+
+        openmc_data_path = _resolve_openmc_data_path(openmc_data_path)
+        data_lib = openmc.data.DataLibrary.from_xml(openmc_data_path)
```

Comment:
renaming to match names used in ```cross_sections ```
```suggestion
        cross_sections = openmc.data.DataLibrary.from_xml(openmc_data_path)
```